--- a/Makefile
+++ b/Makefile
@@ -9,7 +9,7 @@
 OUT := redsocks
 VERSION := 0.5
 
-LIBS := -levent_core
+LIBS := -levent
 ifeq ($(DBG_BUILD),1)
 # -levent_extra is required only for `http` and `debug`
 LIBS += -levent_extra
@@ -26,8 +26,8 @@
 	ctags -R
 
 $(CONF):
-	@case `uname` in \
-	Linux*) \
+	@case `$(CC) -dumpmachine` in \
+	*linux*) \
 		echo "#define USE_IPTABLES" >$(CONF) \
 		;; \
 	OpenBSD) \
@@ -66,7 +66,7 @@
 base.c: $(CONF)
 
 $(DEPS): $(SRCS)
-	gcc -MM $(SRCS) 2>/dev/null >$(DEPS) || \
+	$(CC) -MM $(SRCS) 2>/dev/null >$(DEPS) || \
 	( \
 		for I in $(wildcard *.h); do \
 			export $${I//[-.]/_}_DEPS="`sed '/^\#[ \t]*include \?"\(.*\)".*/!d;s//\1/' $$I`"; \
